tests/testw3emc.f

```diff
@@ -0,0 +1,317 @@
+     &            CTITLE(28:45),CTITLE(46:47),CTITLE(51:62),LGRIB
+               END IF
+            ELSE
+               WRITE(OUTPUT,2802)LABEL(1:5),NPDS,CTITLE(2:26),
+     &         CTITLE(28:45),CTITLE(46:47),CTITLE(51:62),LGRIB
+            ENDIF
+            NS=NS+NLEN
+          ENDDO
+      ELSE
+          PRINT *,' ... COULD NOT READ INDEX FILE = ',CGI(1:NCGI)
+      ENDIF
+      WRITE (OUTPUT,2400) CGI(1:NCGI), NNUM, IARG, TOTAL
+      CALL BACLOSE(32,IRET)
+C
+      CALL W3UTCDAT (ITIME)
+      JREW    = 0
+      JGDS    = -1
+      MPDST   = -1
+      CPARM = 'KWBC'
+      CALL W3AS00(NPARM,CPARM,IER)
+      CPARM = 'KWBC'
+c
+      CALL GETGBP(LUGB,LUGI,MXSIZ3,JREW,MPDST,JGDS,
+     &      itot,KREW,KPDST,KGDS,GRIB,IRET)
+      print *, ' '
+      IF (CHKIRET.EQ.0) THEN
+          print *,'Testing NCEPLIBS-w3emc: SUCCESS !'
+      ELSE
+          print *,'Testing NCEPLIBS-w3emc: FAILED !'
+      ENDIF
+      print *, ' '
+c
+      END
+C-----------------------------------------------------------------------
+      SUBROUTINE WRGI1H(LUGI,NLEN,NNUM,CGB)
+C$$$  SUBPROGRAM DOCUMENTATION BLOCK
+C
+C SUBPROGRAM: WRGI1H         WRITE INDEX HEADERS
+C   PRGMMR: IREDELL          ORG: W/NMC23     DATE: 93-11-22
+C
+C ABSTRACT: THIS SUBPROGRAM WRITES TWO INDEX HEADERS.
+C   CURRENTLY, THE LENGTH OF EACH INDEX RECORD IS 152.
+C
+C PROGRAM HISTORY LOG:
+C   93-11-22  IREDELL
+C   95-10-31  IREDELL   - MODULARIZE SYSTEM CALLS
+C 2012-07-31  VUONG     - CHANGED HOSTNAME TO HOSTNAM
+C
+C USAGE:    CALL WRGI1H(LUGI,NLEN,NNUM,CGB)
+C   INPUT ARGUMENTS:
+C     LUGI         INTEGER LOGICAL UNIT OF OUTPUT INDEX FILE
+C     NLEN         INTEGER LENGTH OF INDEX RECORDS
+C     NNUM         INTEGER NUMBER OF INDEX RECORDS
+C     CGB          CHARACTER NAME OF GRIB FILE
+C
+C SUBPROGRAMS CALLED:
+C   NCBASE       GET BASENAME OF FILE
+C   HOSTNAM      GET SYSTEM NAME
+C   BAWRITE      BYTE-ADDRESSABLE WRITE
+C
+C ATTRIBUTES:
+C   LANGUAGE: FORTRAN
+C
+C$$$
+      CHARACTER CGB*(*)
+      CHARACTER CD8*8,CT10*10,HOSTNAM*15
+      CHARACTER CHEAD(2)*81
+C - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+C  FILL FIRST 81-BYTE HEADER
+      NCGB=LEN(CGB)
+      NCGB1=NCBASE(CGB,NCGB)
+      NCGB2=NCBASE(CGB,NCGB1-2)
+      CALL DATE_AND_TIME(CD8,CT10)
+      CHEAD(1)='!GFHDR!'
+      CHEAD(1)(9:10)=' 1'
+      CHEAD(1)(12:14)='  1'
+      WRITE(CHEAD(1)(16:20),'(I5)') 162
+      CHEAD(1)(22:31)=CD8(1:4)//'-'//CD8(5:6)//'-'//CD8(7:8)
+      CHEAD(1)(33:40)=CT10(1:2)//':'//CT10(3:4)//':'//CT10(5:6)
+      CHEAD(1)(42:47)='GB1IX1'
+      CHEAD(1)(49:54)=CGB(NCGB2:NCGB1-2)
+      CHEAD(1)(56:70)=HOSTNAM(hostname)
+      CHEAD(1)(72:81)='testw3emc '
+      CHEAD(1)(82:82)=CHAR(10)
+C - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+C  FILL SECOND 83-BYTE HEADER
+      CHEAD(2)='IX1FORM:'
+      WRITE(CHEAD(2)(9:38),'(3I10)') 162,NLEN,NNUM
+      CHEAD(2)(41:81)=CGB(NCGB1:NCGB)
+      CHEAD(2)(82:82)=CHAR(10)
```

Comment:
`CHEAD` is a character array that has a length of 81, but you're writing a new line to index 82? That's out-of-bounds.